### PR TITLE
check hex, address and hash

### PR DESCRIPTION
**Pull request by hench-ye**

### CODE REVIEW

File: src/xtopcom/xdata/xconsensus_action.h [hunk 1]
```diff
@@ -373,6 +373,7 @@ class xtop_consensus_action<xtop_action_type_t::evm> : public xtop_action_t<xtop
         }
         m_input_data = top::to_bytes(tx->get_transaction()->get_data());
         m_gaslimit = (uint64_t)tx->get_transaction()->get_gaslimit();
+        xinfo("xtop_consensus_action, m_gaslimit: %llu", m_gaslimit);
```
Comment (jimmy1984xu) [inline src/xtopcom/xdata/xconsensus_action.h:376]: remove this log, xinfo log has performance influence

File: src/xtopcom/xrpc/eth_rpc/eth_method.h [hunk 2]
```diff
@@ -63,24 +63,27 @@ class EthMethod {
...
+        response["result"] = value;
+        //xdbg("web3_clientVersion: %s", response.asString().c_str());
     }
 
-    inline void eth_gasPrice(const Json::Value & request, Json::Value & response) {
-        response = "0x3b9aca00";
-    }
+    void eth_gasPrice(const Json::Value & request, Json::Value & response);
 
     inline void eth_estimateGas(const Json::Value & request, Json::Value & response) {
```
Comment (jimmy1984xu) [inline src/xtopcom/xrpc/eth_rpc/eth_method.h:84]: delete this function, you have already implemented

File: src/xtopcom/xrpc/xrpc_eth_query_manager.cpp [hunk 3]
```diff
@@ -530,51 +597,98 @@ std::string xrpc_eth_query_manager::safe_get_json_value(xJson::Value & js_req, c
...
+    if (!eth::EthErrorCode::check_eth_address(to, js_rsp))
         return;
-    }
     to = xvaccount_t::to_evm_address(to);
     ETH_ADDRESS_CHECK_VALID(to)
 
     std::string from = safe_get_json_value(js_req[0], "from");
-    if (!check_eth_address(from)) {
-        deal_error(js_rsp, eth::enum_invalid_argument_hex);
+    if (from.empty())
```
Comment (jimmy1984xu) [inline src/xtopcom/xrpc/xrpc_eth_query_manager.cpp:612]: can from address be empty ?

File: src/xtopcom/xrpc/xrpc_eth_query_manager.cpp [hunk 4]
```diff
@@ -530,51 +597,98 @@ std::string xrpc_eth_query_manager::safe_get_json_value(xJson::Value & js_req, c
...
+            return;
+    }
+
     top::evm_common::u256 gas_value;
-    if (gas.empty() || gas_price.empty())
-        gas_value = (uint64_t)12000000U;
-    else
-        gas_value = std::strtoul(gas.c_str(), NULL, 16) * std::strtoul(gas_price.c_str(), NULL, 16);
+//    if (gas.empty())
+    gas_value = (uint64_t)12000000U;
```
Comment (jimmy1984xu) [inline src/xtopcom/xrpc/xrpc_eth_query_manager.cpp:645]: the gas value is always 1200000 ?

File: src/xtopcom/xdata/xconsensus_action.h (hunk 1 above)
Comment (hench-ye) [reply src/xtopcom/xdata/xconsensus_action.h]: ok

File: src/xtopcom/xrpc/xrpc_eth_query_manager.cpp [hunk 5]
```diff
@@ -122,18 +106,18 @@ xaccount_ptr_t xrpc_eth_query_manager::query_account_by_number(const std::string
...
     }
 
     base::xauto_ptr<base::xvbstate_t> bstate = base::xvchain_t::instance().get_xstatestore()->get_blkstate_store()->get_block_state(_block.get(), metrics::statestore_access_from_vnodesrv_load_state);
     if (bstate == nullptr) {
         xwarn("xstore::query_account_by_number fail-load state.block=%s", _block->dump().c_str());
-        return nullptr;
+        return enum_block_not_found;
     }
 
     data::xtablestate_ptr_t tablestate = std::make_shared<data::xtable_bstate_t>(bstate.get());
```
Comment (PaytonWu) [inline src/xtopcom/xrpc/xrpc_eth_query_manager.cpp:123]: 这里问题很大，两个智能指针同时管理一个裸指针

File: src/xtopcom/xrpc/xrpc_eth_query_manager.cpp (hunk 4 above)
Comment (hench-ye) [reply src/xtopcom/xrpc/xrpc_eth_query_manager.cpp:645]: std::string msg = std::string("err: intrinsic gas too low: have ") + gas_value.str() + ", want " +
            std::to_string(output.m_tx_result.used_gas) + " (supplied gas " + gas_value.str() + ")";
需要提供这个错误，如果不扩大gas，就没法提供used_gas

File: src/xtopcom/xrpc/xrpc_eth_query_manager.cpp (hunk 3 above)
Comment (hench-ye) [reply src/xtopcom/xrpc/xrpc_eth_query_manager.cpp:612]: yes.

**Review by jimmy1984xu — APPROVED** [review]

**Pull request merged**